controllers/progressivesync_controller.go

```diff
@@ -97,11 +99,16 @@ func (r *ProgressiveSyncReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	latest := ps
```

Comment:
why do we need `latest`? looks like you're constantly reusing it after that, I see no more ref to `ps`

Reply:
line 107